taskfiles/toolchains.yaml

```diff
@@ -1,13 +1,79 @@
 version: "3"
 
+includes:
+  utils: "../tools/yscope-dev-utils/exports/taskfiles/utils/utils.yaml"
+
+vars:
+  G_RUST_TOOLCHAIN_ROOT: "{{.G_BUILD_DIR}}/toolchains/rust"
+
+  # Rust toolchain installation paths
+  G_CARGO_HOME: "{{.G_RUST_TOOLCHAIN_ROOT}}/.cargo"
+  G_RUSTUP_HOME: "{{.G_RUST_TOOLCHAIN_ROOT}}/.rustup"
+
+  # Rust toolchain checksum file
+  G_RUST_TOOLCHAIN_CHECKSUM_FILE: "{{.G_BUILD_DIR}}/toolchains-rust.md5"
+
+  G_RUST_TOOLCHAIN_ENV_FILE: "{{.G_RUST_TOOLCHAIN_ROOT}}/env"
+
 tasks:
   rust:
+    # We use a label to uniquely identify this task and its checksum under `.task/checksums/`.
+    label: "toolchains-rust"
     run: "once"
+    env:
+      CARGO_HOME: "{{.G_CARGO_HOME}}"
+      RUSTUP_HOME: "{{.G_RUSTUP_HOME}}"
+    vars:
+      CHECKSUM_FILE: "{{.G_RUST_TOOLCHAIN_CHECKSUM_FILE}}"
+      # `RUSTUP_HOME` and `CARGO_HOME` contains cache files and temporary files. Instead of
+      # including the entire directory, we only include the necessary files to verify whether the
+      # toolchain installation is up to date.
+      OUTPUT_PATTERNS:
+        - "{{.G_CARGO_HOME}}/bin"
+        - "{{.G_CARGO_HOME}}/env"
+        - "{{.G_RUSTUP_HOME}}/settings.toml"
+        - "{{.G_RUSTUP_HOME}}/toolchains"
+        - "{{.G_RUST_TOOLCHAIN_ENV_FILE}}"
+    sources:
+      - "{{.ROOT_DIR}}/taskfile.yaml"
+      - "{{.TASKFILE}}"
+    generates: ["{{.CHECKSUM_FILE}}"]
+    deps:
+      - task: "utils:checksum:validate"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          INCLUDE_PATTERNS:
+            ref: ".OUTPUT_PATTERNS"
```

Comment:
can we use YAML anchors instead of creating a task variable? e.g., https://github.com/y-scope/clp/blob/dfb4526000bf9d556ac3d1af360a45bf590cbbfc/taskfiles/lint.yaml#L58C15-L70